giraffe-fsharp/Giraffe
# Add F# Analyzers

## Description

I thought it would be beneficial if we were to add some analyzers to the project in order to catch some areas we might improve. This opens up some discussion/thinking for the maintainers to consider configuration and additional pipeline opportunities.

1. Do we want to add analyzers to the build pipeline/PR checks? [See docs](https://ionide.io/FSharp.Analyzers.SDK/content/Running%20during%20CI.html)
2. Are there specific analyzers we want to trigger errors/warning/ignore?

I don't have any strong opinions and just wanted to share some other cool opportunities we could try and add to this PR or others if desired.

## How to test

1. Run `dotnet tool restore`
4. Run `dotnet msbuild /t:AnalyzeSolution`
5. See all the suggestions
  ![image](https://github.com/user-attachments/assets/75dd455e-88d0-441f-b417-5fd55efdfff8)


## Related issues

I was looking through some issues and came across #577 which made me think of suggesting this.
